File: api/app/models/notification.py
import datetime
from typing import Optional
from sqlalchemy import String, DateTime, ForeignKey, Text, func, Boolean
from sqlalchemy.orm import Mapped, mapped_column, relationship
from .base import Base

class Notification(Base):
    __tablename__ = "notifications"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    time_created: Mapped[datetime.datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())
    title: Mapped[str] = mapped_column(String(255), nullable=False)
    noti_message: Mapped[str] = mapped_column(Text, nullable=False)
    is_read: Mapped[bool] = mapped_column(Boolean, nullable=False, default=False)

    usr_id: Mapped[int] = mapped_column(ForeignKey("users.id", ondelete="CASCADE"), nullable=False)
    post_id: Mapped[Optional[int]] = mapped_column(ForeignKey("posts.id", ondelete="CASCADE"))

    user: Mapped["User"] = relationship(back_populates="notifications")
    post: Mapped["Post"] = relationship(back_populates="notifications")

File: schema/notification.ts
import * as z from "zod";

const UserPublicScheme = z.object({
  alias: z.string(),
  followed_threads: z.array(z.any()).optional(),
})

const NotificationSchema = z.object({
  id: z.number(),
  title: z.string(),
  noti_message: z.string(),
  time_created: z.string(),
  is_read: z.boolean(),
  link_to_newpost: z.string().nullable(),
  usr_id: z.number(),
  post_id: z.number().nullable(),
})

export type Notification = z.infer<typeof NotificationSchema>;

export const NotificationListSchema = z.array(NotificationSchema);